tests/overlay-local-store/delete-inner.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+
+set -x
+
+source common.sh
+
+# Avoid store dir being inside sandbox build-dir
+unset NIX_STORE_DIR
+unset NIX_STATE_DIR
+
+storeDirs
+
+initLowerStore
+
+mountOverlayfs
+
+# Add to overlay before lower to ensure file is duplicated
+upperPath=$(nix-store --store "$storeB" --add delete.sh)
+lowerPath=$(nix-store --store "$storeA" --add delete.sh)
+[[ "$upperPath" = "$lowerPath" ]]
+
+# Check there really are two files with different inodes
+upperInode=$(stat -c %i "$storeBRoot/$upperPath")
+lowerInode=$(stat -c %i "$storeA/$lowerPath")
+[[ "$upperInode" != "$lowerInode" ]]
+
+# Now delete file via the overlay store
+nix-store --store "$storeB&remount-hook=$PWD/remount.sh" --delete "$upperPath"
```

Comment:
https://github.com/NixLayeredStore/nix/actions/runs/5669162468/job/15361297296?pr=11#step:6:3925 huh test failure with this.

Reply:
Hmm, because of the `/usr/bin/env` shebang issue you mentioned in the comment below?